vector/src/main/java/im/vector/app/features/ftue/FTUEAction.kt

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2019 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.ftue
+
+import im.vector.app.core.platform.VectorViewModelAction
+import im.vector.app.features.login.LoginConfig
+import im.vector.app.features.login.ServerType
+import im.vector.app.features.login.SignMode
+import org.matrix.android.sdk.api.auth.data.Credentials
+import org.matrix.android.sdk.api.auth.data.SsoIdentityProvider
+import org.matrix.android.sdk.api.auth.registration.RegisterThreePid
+import org.matrix.android.sdk.internal.network.ssl.Fingerprint
+
+sealed class FTUEAction : VectorViewModelAction {
```

Comment:
I think there is a small misunderstanding here, FTUE stands for First Time User Experience, and it does not only cover the login flow, but also the first room creation, the first message sent, etc. So to me the FTUE is the user experience during the (2 or 3) first days using the app.

Also user can create an account or login even if they already know the app (so it will not be the first time)

So using FTUE to name classes related to the signin/signup flow is a bit strange to me.

Reply:
good point, I was considering the FTUE to be the new variant of the onboarding but product/design is calling this the `FTUE authentication flow`

what do you think of using `onboarding` as the package/activity/domain which would then contain `Login, Login2, FTUE Auth`?

Reply:
renamed ftue to onboarding https://github.com/vector-im/element-android/pull/4657/commits/119140b80e9ae61f80fc7eb7242cf8d246d06ce0

we have a onboarding domain with variants `legacy`, `login 2` and `ftue auth`

Reply:
Yes, Onboarding is better, also good for de-jargoning.
By the way, the Java recommendation for acronyms in naming is to capitalize only the first letter of them, so for instance we would have `FtueAction` here, instead of `FTUEAction`.